chanzuckerberg/single-cell-data-portal
# feat: add sqsum array to expression summary default cube

## Reason for Change

first component of this ticket: https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/5093

this PR adds the `sqsum` array to the expression summary default cube. up next:
- actually read from it in `calculate_markers`
- deprecate `expression_summary_fmg`

this PR also includes some aesthetic code changes, such as:
1. ensuring we have the return type annotated on every function i touched
2. adding the `*` decorator to all functions i touched, which requires callers to name the params. this helps ensure that you don't end up touching the wrong parameter just because you ordered something wrong
3. created Result class objects rather than returning tuples. i think having a typed result object is a lot easier to read when you're returning more than two things. similarly, this helps prevent the caller of the function from setting one of the result values to be the wrong thing if you just ordered your variables wrong

## Testing steps

in order to make sure this change works as expected, we need to regenerate our test fixtures. this also means we need to create a new snapshot with the `sqsum` array. i plan to accomplish this by:
1. creating an rdev stack against a commit on this PR
```
happy create $STACK_NAME --tag sha-$COMMIT_TAG --create-tag=false --skip-check-tag
```
2. running the WMG pipeline against that rdev stack
```
aws batch submit-job --job-name wmg-rdev --job-queue wmg-rdev --job-definition wmg-rdev-$STACK_NAME-wmg-processing --profile single-cell-dev
```
3. wait for pipeline to complete... ⏳ about 7 hours
4. download snapshot locally into `../snapshot` (relative to directory for `single-cell-data-portal`). this can be found somewhere in S3
5. generate realistic test fixtures from snapshot
```
python3 scripts/generate_realistic_test_snapshot.py ../snapshot tests/unit/backend/wmg/fixtures/realistic-test-snapshot
```
6. profit
![image](https://github.com/chanzuckerberg/single-cell-data-portal/assets/5653616/93ad8e12-1009-4699-87c9-96fbc045ed67)


## Notes for Reviewer
